packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -442,7 +444,7 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
                                     )
                                 }
                                 val indexableTypes = with(pluginContext.irBuiltIns) {
-                                    setOf(byteType, charType, shortType, intType, longType, stringType, realmInstantType, objectIdType, realmUUIDType).map { it.classifierOrFail }
+                                    setOf(byteType, charType, shortType, intType, longType, stringType, realmInstantType, objectIdType, realmUUIDType, realmMutableIntType).map { it.classifierOrFail }
```

Comment:
I guess this should ideally be using the same infrastructure as the test's `TypeDescriptor` but maybe we could at least for now just move these kind of meta-information to a central place in the compiler plugin source, so that it is not spread across the generators.